lib/EnsEMBL/REST/Controller/Regulatory.pm

```diff
@@ -36,29 +37,131 @@ application/json
 
 BEGIN {extends 'Catalyst::Controller::REST'; }
 
+#protect against missing variables
 
-sub species: Chained('/') PathPart('regulatory') CaptureArgs(1) {
+sub species: Chained('/') PathPart('regulatory/species') CaptureArgs(1) {
   my ( $self, $c, $species) = @_;
   $c->stash(species => $species);
+
+  unless (defined $species) { $c->go('ReturnError', 'custom', [qq{Species must be provided as part of the URL.}])} 
+  $c->stash(species => $species);
+#  $c->log->debug(Dumper "S: $species");
 }
 
-sub id_GET {}
 
+# /regulatory/species/:species/id/:id
+sub id: Chained('species') PathPart('id') Args(1) ActionClass('REST') {
+  my ( $self, $c, $id) = @_;
 
-sub id: Chained('species') PathPart('') Args(1) ActionClass('REST') {
-  my $self = shift;
-  my $c    = shift;
-  my $id   = shift;
+  if(! defined $id){
+    $c->go('ReturnError', 'custom', [qq{Ensembl Stable ID  must be provided as part of the URL.}]);
+  } 
+  if($id !~ /ENSR\d{11}/){
+    $c->go('ReturnError', 'custom', [qq{Ensembl Regulation Stable IDs  have the format ENSR12345678901.}])
+  }
+}
+
+sub id_GET {
+  my ($self, $c, $id) = @_;
   my $regf;
-#  my $pops = $c->request->param('pops');
 
+#  $c->log->debug(Dumper "ID: $id");
   try {
-    $regf = $c->model('Regulatory')->fetch_regulatory($id);
+    $regf = $c->model('Regulatory')->fetch_regulatory_feature($id);
   } catch {
     $c->go('ReturnError', 'from_ensembl', [qq{$_}]) if $_ =~ /STACK/;
     $c->go('ReturnError', 'custom', [qq{$_}]);
   };
   $self->status_ok($c, entity => $regf);
+
+}
+
+# /regulatory/species/:species/epigenome/:epigenome
+sub epigenome: Chained('species') PathPart('epigenome') ActionClass('REST') { }
+
+sub epigenome_GET {
+  my ($self, $c) = @_;
+
+  my $epigenomes;
+  try {
+    $epigenomes = $c->model('Regulatory')->fetch_all_epigenomes();
+  } catch {
+    $c->go('ReturnError', 'from_ensembl', [qq{$_}]) if $_ =~ /STACK/;
+    $c->go('ReturnError', 'custom', [qq{$_}]);
+  };
+  $self->status_ok($c, entity => $epigenomes);
+}
+
+# /regulatory/species/:species/microarray
+sub microarray_list: Chained('species') PathPart('microarray') ActionClass('REST') { }
+sub microarray_list_GET {
+  my ($self, $c) = @_;
+  
+    my $microarrays;
+    try {
+      $microarrays = $c->model('Regulatory')->list_all_microarrays();
+    }catch {
+      $c->go('ReturnError', 'from_ensembl', [qq{$_}]) if $_ =~ /STACK/;
+      $c->go('ReturnError', 'custom', [qq{$_}]);
+    };
+    $self->status_ok($c, entity => $microarrays);
+}
+# stash array name
+sub microarray :Chained('species') PathPart('microarray') CaptureArgs(1) ActionClass('REST') { 
```

Comment:
Strictly speaking you could drop the ActionClass and fold microarray_get() into this method. There's no endpoint to it, so it doesn't need to act like one. Pending it working of course.